lib/services/keyVault/lib/keyVault.js

```diff
@@ -16,366 +16,677 @@
+ * @param {string}                            vault         The vault identifier.
+ * @param {Buffer}                            backupValue   A sequence of bytes obtained through the BACKUP KEY operation.
+ * @param {KeyVaultClient~keyBundleCallback}  callback      A callback that will be called on completion.
+  */
+KeyVaultClient.prototype.restoreKey = function(vault, backupValue, callback) {
+    var keyOpRequest = createRestoreKeyRequest(backupValue);
+    var result = this._keys.restore(vault, serializeKeyRequest(keyOpRequest), keyBundleResponseDeserializer(callback));
+    return result;
 };
 
-function serializeKeyRequest(request) {
-  return {
-    rawJsonRequest: JSON.stringify(request)
-  };
-}
-
-function keyResponseDeserializer(callback) {
-  return function (err, response) {
-    if (err) {
-      return callback(err, null, response);
-    }
-    if (response.keyOpResponse) {
-      return callback(null, JSON.parse(response.keyOpResponse), response);
-    }
-    return callback(new Error('Unrecognized response'), null, response);
-  };
-}
-
-/** A class that performs operations on vault secrets.
-*/
-function KeyVaultSecrets(client) {
-  this.secrets = client.internalClient.secrets;
-}
-
 /**
-* Delete the specified secret
-* @param {string} secretIdentifier
-* @param {function} callback
-* @returns {Stream} The response stream.
-*/
-KeyVaultSecrets.prototype.deleteMethod = function (secretIdentifier, callback) {
-  return this.secrets.deleteMethod(secretIdentifier, secretResponseDeserializer(callback));
+ * Performs a GET SECRET operation, which retrieves secret attributes from the service.
+ * 
+ * @param {string}                              secretIdentifier      The secret identifier. It may or may not contain a version path. If a version is not provided, the latest secret version is used.
+ * @param {KeyVaultClient~secretBundleCallback} callback              A callback that will be called on completion.
+ */
+KeyVaultClient.prototype.getSecret = function(secretIdentifier, callback) {
+    var result = this._secrets.get(secretIdentifier, secretBundleResponseDeserializer(callback));
+    return result;
 };
 
 /**
-* Gets a secret
-* @param {string} secretIdentifier
-* @param {function} callback
-* @returns {Stream} The response stream.
-*/
-KeyVaultSecrets.prototype.get = function (secretIdentifier, callback) {
-  return this.secrets.get(secretIdentifier, secretResponseDeserializer(callback));
+ * Performs a SET SECRET operation, which stores a secret in the service.
+ *
+ * @param {string}                              vault                 The vault identifier.
+ * @param {string}                              secretName            The secret name. If a secret with this name already exists, the service creates a new version and set as current.
+ * @param {string}                              request.value         The secret value.
+ * @param {string}                              [request.contentType] The value's content type.
```

Comment:
Good catch. I renamed the parameter on JsDoc, but forgot to change the implementation. Fixing.